src/dlfcn.c

```diff
@@ -467,7 +472,157 @@ char *dlerror( void )
     return error_buffer;
 }
 
-#ifdef SHARED
+/* taken from http://bandido.ch/programming/Import_Address_Table_Hooking.pdf */
+static IMAGE_IMPORT_DESCRIPTOR* getImportTable( HMODULE module, DWORD *size )
+{
+    IMAGE_DOS_HEADER* dosHeader = (IMAGE_DOS_HEADER*)module;
+    if ( dosHeader->e_magic != 0x5A4D )
+        return NULL;
+    IMAGE_OPTIONAL_HEADER* optionalHeader = (IMAGE_OPTIONAL_HEADER*)
+        ((BYTE*)module + dosHeader->e_lfanew + 24);
+    if ( optionalHeader->Magic != IMAGE_NT_OPTIONAL_HDR_MAGIC )
+        return NULL;
+    if ( optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].Size == 0 ||
+        optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress == 0 )
+        return NULL;
+    *size = optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].Size;
+    return (IMAGE_IMPORT_DESCRIPTOR*)((BYTE*)module +
+        optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress);
+}
+
+/*
+ * return symbol name for a given address
+ */
+static char *getSymbolName( HMODULE baseAddress, IMAGE_IMPORT_DESCRIPTOR *iid, void *addr )
+{
+    void *base = baseAddress;

```

Comment:
So... why do you need `base` variable? Cannot you directly use `baseAddress`?

Reply:
Otherwise the calculations in that function do not work. May be changing the type of parameter `baseAddress` to void * may work. I will try.